### PR TITLE
Change CompositeNode::extension to return a unique_ptr

### PR DESCRIPTION
**Pull request by lukedirtwalker**
That makes much more sense than a pointer since the return pointer is in
most cases deleted after usage. In most cases it was even used as
unique_ptr already and in other cases the pointer leaked.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/dimitar-asenov/Envision/pull/154%23discussion_r40182486%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/154%23issuecomment-142534747%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/154%23discussion_r40182581%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/154%23discussion_r40182630%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/154%23issuecomment-142537562%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%20234b105b9faa5da4c1a663661b28659a2ffe553e%20ModelBase/test/NotificationsTest.cpp%2015%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/154%23discussion_r40182486%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22You%20can%20even%20remove%20this%20line%2C%20it%20doesn%27t%20seem%20to%20be%20needed%20now%20%28it%20was%20used%20for%20deletion%20before%29.%22%2C%20%22created_at%22%3A%20%222015-09-23T09%3A00%3A36Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%2C%20%7B%22body%22%3A%20%22Ah%20I%20thought%20the%20test%20depends%20on%20deletion%20at%20this%20point%22%2C%20%22created_at%22%3A%20%222015-09-23T09%3A01%3A56Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3530769%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/lukedirtwalker%22%7D%7D%2C%20%7B%22body%22%3A%20%22Remove%20it%20and%20run%20the%20test%20to%20make%20sure.%22%2C%20%22created_at%22%3A%20%222015-09-23T09%3A02%3A31Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20ModelBase/test/NotificationsTest.cpp%3AL90-105%22%7D%2C%20%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/154%23issuecomment-142534747%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22This%20looks%20great.%20Thanks%20for%20helping%20out.%22%2C%20%22created_at%22%3A%20%222015-09-23T09%3A01%3A32Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%2C%20%7B%22body%22%3A%20%22Fixed%20comment%20by%20amending%22%2C%20%22created_at%22%3A%20%222015-09-23T09%3A16%3A01Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3530769%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/lukedirtwalker%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [x] <a href='#crh-comment-Pull 234b105b9faa5da4c1a663661b28659a2ffe553e ModelBase/test/NotificationsTest.cpp 15'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/154#discussion_r40182486'>File: ModelBase/test/NotificationsTest.cpp:L90-105</a></b>
- <a href='https://github.com/dimitar-asenov'></a> You can even remove this line, it doesn't seem to be needed now (it was used for deletion before).
- <a href='https://github.com/lukedirtwalker'></a> Ah I thought the test depends on deletion at this point
- <a href='https://github.com/dimitar-asenov'></a> Remove it and run the test to make sure.
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/154#issuecomment-142534747'>General Comment</a></b>
- <a href='https://github.com/dimitar-asenov'></a> This looks great. Thanks for helping out.
- <a href='https://github.com/lukedirtwalker'></a> Fixed comment by amending

<a href='https://www.codereviewhub.com/dimitar-asenov/Envision/pull/154?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/dimitar-asenov/Envision/pull/154?approve=1'></a>&nbsp;<a href='https://github.com/dimitar-asenov/Envision/pull/154'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dimitar-asenov** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/dimitar-asenov/Envision/pull/154'>Refresh</a>.<a href='#crh-update-none'></a>

File: ModelBase/test/NotificationsTest.cpp [hunk 1]
```diff
@@ -90,15 +90,15 @@ TEST(ModelBasePlugin, ModificationNotificationTests)
...
 	CHECK_CONDITION(nl.removedNodes.contains( posLeft->xNode()) );
 	CHECK_CONDITION(nl.removedNodes.contains( posLeft->yNode()) );
 
-	TestNodes::PositionExtension* posRight = right->extension<TestNodes::PositionExtension>();
+	auto posRight = right->extension<TestNodes::PositionExtension>();
 	CHECK_CONDITION(nl.removedNodes.contains( posRight->xNode()) );
 	CHECK_CONDITION(nl.removedNodes.contains( posRight->yNode()) );
 
-	SAFE_DELETE(posLeft);
+	posLeft.reset();
```
Comment (dimitar-asenov) [inline ModelBase/test/NotificationsTest.cpp]: You can even remove this line, it doesn't seem to be needed now (it was used for deletion before).

**Comment by dimitar-asenov** [comment]
This looks great. Thanks for helping out.

File: ModelBase/test/NotificationsTest.cpp (hunk 1 above)
Comment (lukedirtwalker) [inline ModelBase/test/NotificationsTest.cpp]: Ah I thought the test depends on deletion at this point

Comment (dimitar-asenov) [inline ModelBase/test/NotificationsTest.cpp]: Remove it and run the test to make sure.

**Comment by lukedirtwalker** [comment]
Fixed comment by amending

**Pull request merged**